src/OrleansServiceBus/Providers/Streams/EventHub/EventHubDataAdapter.cs

```diff
@@ -138,9 +138,7 @@ public class EventHubDataComparer : ICacheDataComparer<CachedEventHubMessage>
         public int Compare(CachedEventHubMessage cachedMessage, StreamSequenceToken streamToken)
         {
             var realToken = (EventSequenceToken)streamToken;
-            return cachedMessage.SequenceNumber != realToken.SequenceNumber
-                ? (int)(cachedMessage.SequenceNumber - realToken.SequenceNumber)
-                : 0 - realToken.EventIndex;
+            return (int)(cachedMessage.SequenceNumber - realToken.SequenceNumber);
```

Comment:
`EventHubSequenceTokenV2` also has an `EventIndex`. Is that not relevant for comparison?